pinc/filter_project_list.inc

```diff
@@ -11,485 +12,420 @@ include_once($relPath.'misc.inc'); // attr_safe()
 // the field name for the Post Processor is checkedoutby and the field for
 // Project Managers is username.
 
-function process_and_display_project_filter_form($username, $filter_type, $filter_label, $data, $state_sql, $custom_display_fields=NULL)
-{
-    // process the data passed into us for changes
-    $filter_changed=maybe_update_project_filter($username, $filter_type, $data, $state_sql);
-
-    // if the filter was updated or the username is empty, use the new data
-    // to populate the form otherwise pull the data from the stored filter
-    if($filter_changed || $username=="")
-        $form_data_source=$data;
-    else
-        $form_data_source=get_current_project_filter_data($username, $filter_type);
-
-    // display the form
-    display_project_filter_form($filter_type, $filter_label, $form_data_source, $state_sql, $custom_display_fields);
-    display_current_project_filter_settings($username, $filter_type);
-}
+add_script_file("$code_url/scripts/filter_project.js");
```

Comment:
We should not do this. Includes should only add functions and classes to namespaces, not define global variables or have secondary effects. A lot of old DP code doesn't follow this and we're slowly working to fix it.

Pages should be explicit about what javascript and other resources they are using.

Reply:
I was uncertain about whether this was a good idea as I said above. I'll change to doing it the normal way.
And write a description.

Reply:
In fact my function would cause the js file to be included in activity_hub where it is not needed so I see the point of the php files including js files explicitly.